### DISCUSSION TITLE
Go module version number

### DISCUSSION THREAD

**Discussion by ashep** [zitadel/zitadel#6920] [Q&A] [answered]
Hello there! According to Go's [documentation](https://go.dev/doc/modules/release-workflow), the version number should be added to a module path in case if major module's version is above 1. Currently, Zitdel's [go.mod file](https://github.com/zitadel/zitadel/blob/main/go.mod) of the latest 2.x version does not contain the version suffix, which leads to the following issue:

```
go get -v github.com/zitadel/zitadel@v2.41.1
go: github.com/zitadel/zitadel@v2.41.1: invalid version: module contains a go.mod file, so module path must match major version ("github.com/zitadel/zitadel/v2")
```

The question is how to avoid the error above while `go get`ting the module?

Thank you in advance.

**Comment by adlerhurst** [comment]
yes you are right, why do you want to go get zitadel? I'm asking because we provide client libraries like [zitadel-go](https://github.com/zitadel/zitadel-go) if you want to call zitadel.

**Reply by ashep** [reply]
Thank you for the answer, @adlerhurst! I'm trying to make a service that could embed Zitadel into its binary and run the Zitadel server in the same process, that's why I want to `go get github.com/zitadel/zitadel`. Could you please help me understand why Zitadel's module path wasn't changed after switching to version 2.x and whether it's possible to fix this?

**Reply by adlerhurst** [reply, marked as answer]
[An issue](https://github.com/zitadel/zitadel/issues/5575) was raised for tracking. users of zitadel normally don't face this problem because they generate the client themselves or use client libraries.

I tried out what would happen if we do the changes in [this PR](https://github.com/zitadel/zitadel/pull/6923) and the current changes would break the go clients which has to be solved first and takes more time than we can provide in the current sprint.

**Reply by ashep** [reply]
Thank you for the answer @adlerhurst!
